configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -0,0 +1,386 @@
+
+                if (hasInitialValue(fieldDeclaration)) {
+                    FieldDescriptor fd = new FieldDescriptor(
+                            fieldDeclaration.getVariable(0).
+                                    getInitializer().get().
+                                    toStringLiteralExpr().get().
+                                    getValue(),
+                            fieldDeclaration.getElementType().toString(),
+                            null, null);
+                    fieldDeclaration.getComment().ifPresent(c ->
+                            fd.comments = cleanComment(c.getContent()));
+                    arg.add(fd);
+                }
+            }
+        }, fields);
+    }
+
+
+    private void addFieldsFromTypeElement(TypeElement initParams,
+                                          List<FieldDescriptor> fields) {
+        processingEnv.getElementUtils().getAllMembers(initParams).stream()
+                .filter(m -> ElementKind.FIELD.equals(m.getKind()))
+                .filter(m -> m instanceof VariableElement) // in theory all fields are VariableElements
+                .map(m -> (VariableElement) m)
+                .map(m -> new FieldDescriptor(m.getSimpleName().toString(),
+                        m.asType().toString(),
+                        getInitialValueAssignmentCode(m),
+                        processingEnv.getElementUtils().getDocComment(m)))
+                .forEach(fields::add);
+    }
+
+    private String getInitialValueAssignmentCode(VariableElement m) {
+        String className = m.getEnclosingElement().toString();
+
+        CompilationUnit cu = getCompilationUnit(className);
+
+        String value = getInitialValueAssignmentCode(cu, m.getSimpleName().toString());
+
+        return value;
+    }
+
+    private CompilationUnit getCompilationUnit(String className) {
+        return compilationUnitCache.computeIfAbsent(className, key -> {
+            String pkgName = key.substring(0, key.lastIndexOf('.'));
+            String simpleClassName = key.substring(key.lastIndexOf('.') + 1);
+            CompilationUnit cu = null;
+            try {
+                FileObject source = processingEnv.getFiler()
+                        .getResource(StandardLocation.SOURCE_PATH, pkgName,
+                                simpleClassName + ".java");
+                ParseResult<CompilationUnit> parsed = new JavaParser()
+                        .parse(source.openInputStream());
+                if (parsed.getResult().isPresent()) {
+                    cu = parsed.getResult().get();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return cu;
+        });
+    }
+
+    private String getInitialValueAssignmentCode(CompilationUnit cu, String fieldName) {
+        assert fieldName != null;
+
+            String[] value = new String[1];
+            if (cu != null) {
+                cu.accept(new VoidVisitorAdapter<String[]>() {
```

Comment:
Just following the visitor pattern as expected. Not sure how to avoid using some value holder (e.g `String[]`) by introducing a nested class here.

Anyway, I will change the `arg` name